sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -170,7 +179,8 @@ private[sql] case class PhysicalRDD(
       |     if ($idx == 0) $numOutputRows.add(numRows);
       |
       |     while (!shouldStop() && $idx < numRows) {
-      |       InternalRow $row = $batch.getRow($idx++);
+      |       int $rowidx = $idx++;
+      |       ${colDeclarations.mkString("", "\n", "\n")}
```

Comment:
Move this to before the while loop (the contract is that the batch object is reused)